### PR TITLE
Add sleep timer

### LINKED ISSUE #105: Add a timer to shut down the app after some time

### PR DESCRIPTION
**Pull request by esensar**
This closes #105

## TODO
- [x] Wait for https://github.com/SimpleMobileTools/Simple-Commons/pull/1815 and update Simple-Commons ref

### CODE REVIEW

**Comment by tibbi** [comment]
you should have moved the sleep_timer string into Commons, avoid duplicating strings

**Comment by tibbi** [comment]
lets add a radiogroup at Custom, with Minutes or Seconds. You can reuse the dialog with Custom repeatition interval from the calendar

**Comment by tibbi** [comment]
and colors of the sleep timer dont change properly on resume

**Comment by tibbi** [comment]
also add lower default values, like 1 and 2 minutes

**Comment by tibbi** [comment]
it should be 1 "minute", not "minutes"

**Comment by tibbi** [comment]
the sleep timer doesnt show up well in Bright Display mode, it is behind the navigation bar. And it should close the whole app, not just go back to the main screen.

**Comment by tibbi** [comment]
lets actually add lower default values, specifically like 10 seconds, 30 seconds, 1 minute, 5 minutes, 10 minutes and 30 minutes

File: app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt [hunk 1]
```diff
@@ -306,9 +306,10 @@ class MainActivity : SimpleActivity() {
         }
         val minutes = getString(R.string.minutes_raw)
         val hour = resources.getQuantityString(R.plurals.hours, 1, 1)
+        val minute = resources.getQuantityString(R.plurals.minutes, 1, 1)
 
         val items = arrayListOf(
-            RadioItem(1 * 60, "1 $minutes"),
+            RadioItem(1 * 60, "1 $minute"),
```
[the remark this answers is not in the document]
Comment (esensar) [reply app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt:312]: Yeah, just realized I can clean this up to use quantity string for all of these items.

**Comment by tibbi** [comment]
app still isnt killed with bright display, but thats fine and maybe isnt even doable without some extra hacks, so it is good enough for now. Thanks

**Pull request merged**